libs/core/kiln_ai/tools/kiln_task_tool.py

```diff
@@ -0,0 +1,183 @@
+                    "model_provider": run_config.run_config_properties.model_provider_name,
+                    "adapter_name": adapter.adapter_name(),
+                },
+                run_config=run_config.run_config_properties,
+            ),
+            output_source_type=DataSourceType.tool_call,
+        )
+
+        # Return structured information about the created task as tool run
+        return json.dumps(
+            {
+                "output": task_run.output.output,
+                "task_run_id": task_run.id,
+                "task_id": task_run.parent.id if task_run.parent else None,
+                "project_id": self._project_id,
+                "tool_id": self._tool_id,
+            }
+        )
+
+    async def _get_task(self) -> Task:
+        """Lazy load the task."""
+        if self._task is None:
+            # Load the project first
+            project = project_from_id(self._project_id)
+            if project is None:
+                raise ValueError(f"Project not found: {self._project_id}")
+
+            # Load the task from the project
+            self._task = Task.from_id_and_parent_path(self._task_id, project.path)
+            if self._task is None:
+                raise ValueError(
+                    f"Task not found: {self._task_id} in project {self._project_id}"
+                )
+        return self._task
+
+    async def _get_run_config(self) -> TaskRunConfig:
+        """Lazy load the task run config."""
+        if self._run_config is None:
+            task = await self._get_task()
+            self._run_config = next(
+                (
+                    run_config
+                    for run_config in task.run_configs(readonly=True)
+                    if run_config.id == self._run_config_id
+                ),
+                None,
+            )
+            if self._run_config is None:
+                raise ValueError(
+                    f"Task run config not found: {self._run_config_id} for task {self._task_id} in project {self._project_id}"
+                )
+        return self._run_config
+
+    async def get_parameters_schema(self) -> Dict[str, Any]:
+        """Lazy load the task parameters schema."""
+        if self._parameters_schema is None:
+            task = await self._get_task()
+
+            # V2: The user should control the input schema, especially for plaintext tasks.
+            if task.input_json_schema:
+                # Use the task's input schema directly if it exists
+                self._parameters_schema = task.input_schema()
+            else:
+                # For plaintext tasks, create a simple string input parameter
+                self._parameters_schema = {
+                    "type": "object",
+                    "properties": {
+                        "input": {
+                            "type": "string",
+                            "description": f"Plaintext input for the task: {task.instruction}",
```

Comment:
Critical: task.instructions could be 5000 words. Way too long and unbounded for a tool description. It's also a prompt, which isn't the right thing to put in a description. 

Just "input to the tool" is fine here I think (task is Kiln specific word, model sees tools). The task description can be tuned, and this should work as well as any general purpose string.